core/src/main/java/org/kiwix/kiwixmobile/core/utils/ConnectivityReporter.kt

```diff
@@ -23,9 +23,14 @@ import java.lang.reflect.InvocationTargetException
 import java.lang.reflect.Method
 import javax.inject.Inject
 
-class WifiManagerUtils @Inject constructor(private val wifiManager: WifiManager) {
+class ConnectivityReporter @Inject constructor(private val wifiManager: WifiManager) {
 
-  fun checkWifi(): Boolean = wifiManager.isWifiEnabled
+  fun checkWifi(): Boolean = if (wifiManager.isWifiEnabled) {
+    val wifiInfo = wifiManager.connectionInfo
```

Comment:
maybe add `connectivityManager` and `connectivityManager.networkState==CONNECTED` might give us the same result a bit conciser.

or rewrite to `= wifiManager.isWifiEnabled & wifiManager.connectionInfo.networkId != -1`

Reply:
@macgills done.